packages/react/src/components/DataTable/DataTable.js

```diff
@@ -392,15 +411,18 @@ export default class DataTable extends React.Component {
    * @param {object} initialState
    * @returns {object} object to put into this.setState (use spread operator)
    */
-  setAllSelectedState = (initialState, isSelected) => {
+  setAllSelectedState = (initialState, isSelected, filteredRowIds) => {
     const { rowIds } = initialState;
     return {
       rowsById: rowIds.reduce(
         (acc, id) => ({
           ...acc,
           [id]: {
             ...initialState.rowsById[id],
-            isSelected: initialState.rowsById[id].disabled ? false : isSelected,
+            isSelected:
+              initialState.rowsById[id].disabled || !filteredRowIds.includes(id)
```

Comment:
would this have the same effect? I think we may be able to avoid ternary here

`(!initialState.rowsById[id].disabled || filteredRowIds.includes(id)) || isSelected`

Reply:
Should be able to avoid the ternary with `(!initialState.rowsById[id].disabled && filteredRowIds.includes(id) && isSelected)`